### PR TITLE
[DI] Autowiring and factories are incompatible with each others

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 2.8
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php [hunk 1]
```diff
@@ -284,6 +284,10 @@ private function getReflectionClass($id, Definition $definition)
             return $this->reflectionClasses[$id];
         }
 
+        if ($definition->getFactory() || $definition->getFactoryClass(false) || $definition->getFactoryService(false)) {
```
Comment (stof) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: shouldn't this be moved at the beginning of `completeDefinition` instead ?

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: true, done

**Review by stof — APPROVED** [review]
👍

**Review by GuilhemN — APPROVED** [review]

**Review by dunglas — APPROVED** [review]
👍

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**